installer/upgrader/Migrations/V5/LangStringHelper.php

```diff
@@ -0,0 +1,228 @@
+            ->executeQuery();
+
+        $query2 = $this->createQueryBuilder();
+        $query2->delete('ohrm_i18n_lang_string')
+            ->andWhere($query2->expr()->in('ohrm_i18n_lang_string.id', ':deleteIds'))
+            ->setParameter('deleteIds', $deleteStrings, Connection::PARAM_INT_ARRAY)
+            ->executeQuery();
+    }
+
+    /**
+     * @param int $groupId
+     * @return array
+     * @throws Exception
+     */
+    private function getNonCustomLangStringIds(int $groupId): array
+    {
+        $query = $this->createQueryBuilder();
+        $query->select('translate.lang_string_id')
+            ->from('ohrm_i18n_lang_string', 'langString')
+            ->leftJoin('langString', 'ohrm_i18n_translate', 'translate', 'langString.id = translate.lang_string_id')
+            ->where('langString.group_id = :module')
+            ->andWhere('translate.customized = 1')
+            ->setParameter('module', $groupId);
+        $results = $query->executeQuery()->fetchAllAssociative();
+
+        $customStrings = array_column($results, 'lang_string_id');
+
+        $query2 = $this->createQueryBuilder();
+        $query2->select('langString.id')
+            ->from('ohrm_i18n_lang_string', 'langString')
+            ->andWhere($query2->expr()->notIn('langString.id', ':customStrings'))
+            ->andWhere('langString.group_id = :module')
+            ->setParameter('customStrings', $customStrings, Connection::PARAM_INT_ARRAY)
+            ->setParameter('module', $groupId);
+        $results2 = $query2->executeQuery()->fetchAllAssociative();
+
+        return array_column($results2, 'id');
+    }
+
+    /**
+     * @param array $langStringArray
+     * @param int $groupId
+     * @return void
+     * @throws Exception
+     */
+    public function versionMigrateLangStrings(array $langStringArray, int $groupId)
+    {
+        foreach ($langStringArray as $langString) {
+            $result = $this->getLangStringRecord($langString->getValue(), $groupId);
+            if ($result == null) {
+                $langStringObj = new LangString(
+                    $langString->getUnitId(),
+                    $langString->getGroupId(),
+                    $langString->getValue(),
+                    $langString->getVersion(),
+                    $langString->getNote()
+                );
+                $this->saveLangString($langStringObj);
+            } else {
+                $this->updateLangStrings($langString->getUnitId(), array_column($result, 'id'));
+            }
+        }
+    }
+
+    /**
+     * @param string $langStringValue
+     * @param int $groupId
+     * @return array
+     * @throws Exception
+     */
+    private function getLangStringRecord(string $langStringValue,int $groupId): array
+    {
+        $q = $this->createQueryBuilder();
+        $q->select('langString.id')
+            ->from('ohrm_i18n_lang_string', 'langString')
+            ->where('langString.value = :source')
```

Comment:
Need to improve this to handle parameters in strings
4.x - %param%
5.x - {param}

Reply:
will handle this in the next improvement now that we will have the ymls by then too